editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -1245,10 +1349,156 @@ void SpriteFramesEditor::drop_data_fw(const Point2 &p_point, const Variant &p_da
 	}
 }
 
+void SpriteFramesEditor::_update_stop_icon() {
+	bool is_playing = false;
+	if (animated_sprite) {
+		is_playing = animated_sprite->call("is_playing");
+	}
+	if (is_playing) {
+		stop->set_icon(pause_icon);
+	} else {
+		stop->set_icon(stop_icon);
+	}
+}
+
+void SpriteFramesEditor::_remove_sprite_node() {
+	if (!animated_sprite) {
+		return;
+	}
+	if (animated_sprite->is_connected("sprite_frames_changed", callable_mp(this, &SpriteFramesEditor::_edit))) {
+		animated_sprite->disconnect("sprite_frames_changed", callable_mp(this, &SpriteFramesEditor::_edit));
+	}
+	if (animated_sprite->is_connected("animation_changed", callable_mp(this, &SpriteFramesEditor::_sync_animation))) {
+		animated_sprite->disconnect("animation_changed", callable_mp(this, &SpriteFramesEditor::_sync_animation));
+	}
+	if (animated_sprite->is_connected("animation_finished", callable_mp(this, &SpriteFramesEditor::_update_stop_icon))) {
+		animated_sprite->disconnect("animation_finished", callable_mp(this, &SpriteFramesEditor::_update_stop_icon));
+	}
+	animated_sprite = nullptr;
+}
+
+void SpriteFramesEditor::_fetch_sprite_node() {
+	Node *selected = nullptr;
+	EditorSelection *editor_selection = EditorNode::get_singleton()->get_editor_selection();
+	if (editor_selection->get_selected_node_list().size() == 1) {
+		selected = editor_selection->get_selected_node_list()[0];
+	}
+
+	bool show_node_edit = false;
+	AnimatedSprite2D *as2d = Object::cast_to<AnimatedSprite2D>(selected);
+	AnimatedSprite3D *as3d = Object::cast_to<AnimatedSprite3D>(selected);
+	if (as2d || as3d) {
+		if (frames != selected->call("get_sprite_frames")) {
+			_remove_sprite_node();
+		} else {
+			animated_sprite = selected;
+			if (!animated_sprite->is_connected("sprite_frames_changed", callable_mp(this, &SpriteFramesEditor::_edit))) {
+				animated_sprite->connect("sprite_frames_changed", callable_mp(this, &SpriteFramesEditor::_edit));
+			}
+			if (!animated_sprite->is_connected("animation_changed", callable_mp(this, &SpriteFramesEditor::_sync_animation))) {
+				animated_sprite->connect("animation_changed", callable_mp(this, &SpriteFramesEditor::_sync_animation), CONNECT_DEFERRED);
+			}
+			if (!animated_sprite->is_connected("animation_finished", callable_mp(this, &SpriteFramesEditor::_update_stop_icon))) {
+				animated_sprite->connect("animation_finished", callable_mp(this, &SpriteFramesEditor::_update_stop_icon));
+			}
+			show_node_edit = true;
+		}
+	} else {
+		_remove_sprite_node();
+	}
+
+	if (show_node_edit) {
+		_sync_animation();
+		autoplay_container->show();
+		playback_container->show();
+	} else {
+		_update_library(); // To init autoplay icon.
+		autoplay_container->hide();
+		playback_container->hide();
+	}
+}
+
+void SpriteFramesEditor::_play_pressed() {
+	if (animated_sprite) {
+		animated_sprite->call("stop");
+		animated_sprite->call("play", animated_sprite->call("get_animation"));
```

Comment:
Here and elsewhere. Why do you `call("stop")` instead of `stop()`? For GDScript "overrides" to be called? But it looks like overriding non-virtual methods shouldn't work. See #55024.

Reply:
AnimatedSprite2D and AnimatedSprite3D have different base class, so we need to "call" method to call same method in them.